From c85408c5f41afb58c72f1a3c6f475ff432ece603 Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Sat, 20 Apr 2019 13:04:05 +0200
Subject: [PATCH 34/75] genetlink: annotate genl_{lock,unlock}

This handles:

net/netlink/genetlink.c:34:1: warning: mutex 'genl_mutex' is still held at the end of function [-Wt
hread-safety-analysis]
}
^
net/netlink/genetlink.c:33:2: note: mutex acquired here
        mutex_lock(&genl_mutex);
        ^
net/netlink/genetlink.c:39:2: warning: releasing mutex 'genl_mutex' that was not held [-Wthread-safety-analysis]
        mutex_unlock(&genl_mutex);
        ^
2 warnings generated.

This causes a number of new warnings.
---
 net/netlink/genetlink.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/netlink/genetlink.c b/net/netlink/genetlink.c
index f0ec068e1d02..e17fd53cc465 100644
--- a/net/netlink/genetlink.c
+++ b/net/netlink/genetlink.c
@@ -28,13 +28,13 @@ static DECLARE_RWSEM(cb_lock);
 atomic_t genl_sk_destructing_cnt = ATOMIC_INIT(0);
 DECLARE_WAIT_QUEUE_HEAD(genl_sk_destructing_waitq);
 
-void genl_lock(void)
+void genl_lock(void) __attribute__((acquire_capability(genl_mutex)))
 {
 	mutex_lock(&genl_mutex);
 }
 EXPORT_SYMBOL(genl_lock);
 
-void genl_unlock(void)
+void genl_unlock(void) __attribute__((release_capability(genl_mutex)))
 {
 	mutex_unlock(&genl_mutex);
 }
-- 
2.17.1

